plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/register-product-collection.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { BlockVariation, registerBlockVariation } from '@wordpress/blocks';
+import { addFilter } from '@wordpress/hooks';
+import { EditorBlock } from '@woocommerce/types';
+import type { ElementType } from '@wordpress/element';
+import type { BlockEditProps } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import blockJson from '../block.json';
+import {
+	SetPreviewState,
+	PreviewState,
+	ProductCollectionAttributes,
+} from '../types';
+
+export interface ProductCollectionConfig extends BlockVariation {
+	preview?: {
+		setPreviewState?: SetPreviewState;
+		initialPreviewState?: PreviewState;
+	};
+}
+
+/**
+ * Register a new collection for the Product Collection block.
+ *
+ * @param {ProductCollectionConfig} blockVariationArgs The configuration of new collection.
+ */
+const registerProductCollection = ( {
```

Comment:
Flagging this for the next work of exposing this function that we'll need to think what arguments will be optional, like handling the `DEFAULT_QUERY` or `DEFAULT_ATTRIBUTES` vars.

Reply:
Thanks for flagging this. I agree that we will need to think about it. For now, I am thinking that we will use DEFAULT values for attributes that aren't provided by 3PDs. But we can definitely think about it in detail while exposing this function, as there might be edge cases that I haven't considered yet.